--- freecad-0.15.4671/src/Mod/Drawing/App/FeaturePage.cpp.orig	2015-04-05 18:25:02.000000000 +0100
+++ freecad-0.15.4671/src/Mod/Drawing/App/FeaturePage.cpp	2016-01-28 22:47:58.404896737 +0000
@@ -195,7 +195,7 @@
             if (count < editText.size()) {
                 // change values of editable texts
                 boost::regex e2 ("(<text.*?freecad:editable=\""+what[1].str()+"\".*?<tspan.*?)>(.*?)(</tspan>)");
-                boost::re_detail::string_out_iterator<std::string > out(newfragment);
+                std::back_insert_iterator<std::string> out(newfragment);
                 boost::regex_replace(out, begin, what[0].second, e2, "$1>"+editText[count]+"$3");
             }
             count++;
